### ISSUE TITLE
Brainstorm - Mentoria no Front

### ISSUE THREAD

**Issue by baltazarparra** [frontendbr/open-source#5]
## Mentoria no Front
Discussão iniciada na issue [#108](https://github.com/frontendbr/forum/issues/108#issuecomment-206926284)

## Objetivo
Brainstorm visando startar um projeto para auxiliar novos desenvolvedores,
onde um mentor ajudaria com sua experiência/visão a resolver conflitos/questões relacionadas aos caminhos que o desenvolvedor pode ou não seguir.

**Comment by woliveiras** [comment]
Cara, eu adoro fazer isso! 

Eu gostaria de ajudar.

**Comment by robsonbittencourt** [comment]
Eu não sei como, mas gostaria de ajudar.

**Comment by woliveiras** [comment]
Ajudo um pessoal pelo Facebook, email e no Slack, quando perguntam sobre carreira.

Carreira é algo que eu realmente curto falar!

Tenho até uns posts falando do assunto:

http://woliveiras.com.br/posts/um-ano-como-desenvolvedor-front-end/
http://woliveiras.com.br/posts/requisitos-para-vagas-front-end/
http://woliveiras.com.br/posts/guia-de-estudos-desenvolvedor-front-end-iniciante/

Se me acharem apto, gostaria de ajudar como mentor.

Trabalhei numa empresa onde aprendi sobre Plano de Carreira e recebi Coach de um profissional de RH e Psicólogo. Deu pra aprender muito e gostaria de repassar esse conhecimento pra quem está iniciando.

Parabéns pela iniciativa e ideia @baltazarparra

**Comment by baltazarparra** [comment]
Legal @woliveiras você pode nos ajudar a moldar então o projeto.

Pontos a se pensar:

Montaríamos pequenos grupos de desenvolvedores que ficariam sobre a "tutela" de um mentor, por um determinado espaço de tempo?

Vamos seccionar os grupos por nível de desenvolvedores pra ficar mais fácil pro mentor?

O mentor vai ter algum tipo de renda vinda desse projeto?

**Comment by LFeh** [comment]
Vejo essa ideia e da issue #3 com certas similaridades. Não rola um merge nas ideias?

**Comment by baltazarparra** [comment]
Como o @woliveiras pilhou nesse projeto, acho que sim, vamos alinhar.

**Comment by woliveiras** [comment]
Verdade @LFeh 

Vamos planejar sabagaçae!

**Comment by baltazarparra** [comment]
@woliveiras vamos jogar essa projeto dentro do seu CT? Talvez convocar uns 3~5 mentores pra ajudar a moldar como isso vai funcionar de fato.

**Comment by danilovaz** [comment]
Acho interessante essa ideia de mentoria. Queria deixar aqui minha opinião, se for de ajuda:

Creio que seria interessante definir que cada mentor terá um pequeno grupo, acredito que talvez no máximo até 10 pessoas (?) dessa maneira o objetivo de auxiliar novos desenvolvedores pode ser atingido mais rápido e com mais qualidade.

Se for deixar 1 mentor com sei lá, 50 desenvolvedores, além do próprio mentor surtar, é bem provável que não se alcance o objetivo. Com muitas pessoas fica difícil dar a atenção devida. 

É a mesma coisa de entrar em um curso de inglês, por exemplo, onde existem 25 alunos na sala de aula. O tempo para você aprender é maior, porque a atenção do professor é dividida. Fiz um curso em uma escola de inglês na cidade onde moro, que seguia uma metodologia de mnemônica e neurolinguística; e para complementar isso eles tinham como regra que cada aula poderia ter no máximo 5 alunos. Assim a atenção tanto do professor como dos demais alunos não era prejudicada e o rendimento era maior.

Não sei se talvez eu não compreendi o intuito, mas eu acredito que em um formato nesse tipo de 1 mentor com no máximo até X desenvolvedores  possa ser melhor proveitoso para os alunos e não acaba sendo uma sangria para o mentor. Abre espaço também para criar grupos que contém mentores com o mesmo propósito, para poder suprir o número de pessoas interessadas na mentoria.

**Comment by baltazarparra** [comment]
Concordo com um numero limitado, com certeza vai ser mais proveitoso, algo entre 5 e 10 desenvolvedores. E acredito que cada desenvolvedor deverá colaborar com uma quantia "X" pro mentor ter algum tipo de retorno também.

**Comment by danilovaz** [comment]
@baltazarparra acredito que tudo que envolva dinheiro, é delicado, por N motivos. Por isso, acho que se for definido algo do tipo: "Para você receber a mentoria de Fulano, você TEM que pagar X por mês". Vai acabar afastando algumas pessoas. É aquele negócio, as pessoas gostam de ter mais direitos do que deveres. Mas também, tem a parte de que tempo é dinheiro e todo mundo precisa comer; dito isso, por que não algo tipo o Patreon? Se for baseado em doação, a aderência talvez seja maior e mais fiél. Afinal, somos uma comunidade, um ajuda o outro. O cara tá me ajudando com mentoria no tempo "livre" dele e eu ajudo como puder financeiramente e SE eu puder. Afinal de contas, não é todo dev que tá empregado ou ganhando bem ou o suficiente. Então se for algo no estilo Patreon, que seja **opcional** a **doação** para o mentor além de ficar mais transparente, acredito que acaba gerando uma adesão bem legal.

**Comment by baltazarparra** [comment]
@danilovaz sim sim, tipo, eu dei uma ideia só, mas acredito que trabalhar com dinheiro sempre é delicado de fato! Se conseguirmos chegar em senso, de outra forma, Ótimo.

**Comment by danilovaz** [comment]
@baltazarparra sim, a ideia é bem legal. e de fato, tem que ter algum retorno/incentivo ao mentor. é algo a ir se discutindo também

**Comment by baltazarparra** [comment]
Então, os pontos que já temos, são:

- Um grupo limitado, de 5 a 10 pessoas.
- Pensar na melhor forma de retorno para o Mentor.
...

**Comment by augustoabreu** [comment]
Por que não deixar que o Mentor escolha a quantidade de pessoas do grupo? Aliás, como classificar uma pessoa como Mentor?

**Comment by baltazarparra** [comment]
@augustoabreu ponto vital para o projeto fluir...
Alguém tem alguma ideia?

**Comment by danilovaz** [comment]
@augustoabreu acho que limitar os grupos é benéfico pro bom funcionamento da mentoria como um todo. acaba permitindo uma melhor organização e atenção por parte do mentor.

**Comment by danilovaz** [comment]
@baltazarparra a grande questão é: como organizar isso?

Podemos por exemplo, criar um repositório chamado Mentorias e organizar as mentorias por issues?

Ou seria melhor criar um canal mais "abrangente"? Com um site, login, canal direto entre o Mentor e o Aprendiz?

**Comment by baltazarparra** [comment]
Acho legal deixar tudo aberto pelo github mesmo, de forma organizada, claro.

**Comment by danilovaz** [comment]
@baltazarparra nesse caso, uma boa organização de issues (como já é feito aqui) fica top.

Imaginei algo assim, veja o que acha: 

Um repositório chamado Mentorias. Uma "fix" issue onde todos  que quiserem ser mentores, devem responde a issue conforme um issue_template.md

Ex.
**Nome do Mentor**

**Área de Mentoria**

**Referências na Área** *(para credibilidade?)*

Link da Issue do Mentor

Aí o mentor deve criar uma issue, talvez, nesse estilo: **[MENTOR] Nome do Mentor - Área de mentoria**

Dentro dessa issue deve conter as seguintes informações

**Descrição da área de mentoria**

**Experiência e conhecimentos do mentor**

**Pontos em que o mentor julgar ser útil em ajudar**

**Contato** *(contatos que o mentor deixa disponível para os aprendizes, fica a critério de cada um, um slack, twitter, facebook, email, whatsapp, whatever)*

Dentro dessa issue do Mentor todos que queiram ser aprendizes, fazem as perguntas e o Mentor responde ali para que todos tenham fácil acesso às respostas.

**Comment by baltazarparra** [comment]
Acho que assim, vai ficar da forma como o forum já é hoje.

A estrutura está ótima, mas precisamos "ligar" mais o mentor, com o grupo de desenvolvedores sobre sua "tutela" de alguma forma. Talvez um gitter com o grupo e o mentor.

Esse template que você criou, pode ser muito útil pra fechar a turma com o mentor.

**Comment by danilovaz** [comment]
é, tem razão. o gitter é muito bom nesse propósito

**Comment by woliveiras** [comment]
Galera, não deu pra acompanhar em real time, mas os pontos que peguei lendo agora foram:

**Doação opcional**, `perfeito`. Se o cara pode doar, ele doa, senão o mentor ajuda na faixa.
 Temos de ter em mente que queremos fazer algo para ajudar iniciantes e, nem sempre, o iniciante tem grana, como o @danilovaz falou. Eu gostaria muito de ajudar, pois, no começo da minha carreira, eu tive esse tipo de ajuda e foi muito bom! Eu recebi ajuda gratuita e nem teria grana pra pagar na época, então eu curtiria se fosse tudo free, porém uma doação opcional é perfeito pra quem pode doar incentivar o mentor.

**Quantidade de pupilos**, acho que 10 por mentor é perfeito, pois muita gente o cara pode perder o foco ou mesmo ficar focado demais. Temos de lembrar que a galera precisa ter vida fora da internet, apesar de nossos projetos para ajudar um ao outro.

**Como definir quem pode ser mentor**, esse ponto é bem foda. Acho que tem de avaliar o perfil da pessoa, não somente perfil técnico, pois um cara que não tem paciência, não consegue ajudar uma pessoa iniciante. Vocês acham que tempo de experiência do mentor deve contar nessa hora ou somente o conhecimento do mesmo?
Eu mesmo gostaria muito de ser mentor, porém tenho um pouco mais de um ano na área como Front End (passei a vida nos servidores gente, não me julguem! ;P).
Que tal levantar uma galera que quer ser mentor primeiro, depois fazemos uma bancada pra escolher quem vai ser?
Igual acontece nos Call4Papers dos eventos. :)

**Comment by woliveiras** [comment]
Sobre mergear no CT, @baltazarparra, da pra criar um repo lá pra gente trabalhar tudo sobre mentoria em cima dele, que tal?

Poderíamos até continuar a discussão direto no repo "mentoria" se quiserem tratar direto no projeto.

**Comment by woliveiras** [comment]
Depois de definidos os mentores, podemos fazer um arquivo ou TypeForm da vida com as perguntas iniciais pros pupilos responderem antes mesmo de entrarem em contato com os caras.

Se acharem interessante a conversa mentor > pupilo, pupilo > mentor, ser em issues, da pra trabalhar lá, mas algumas perguntas as pessoas tem vergonha de fazer em público por achar que são perguntas tolas. Poderia ser direto num email do mentor também.

**Comment by danilovaz** [comment]
**Como definir quem pode ser mentor**

Você tocou em um ponto importantíssimo. Eu acho que tão importante quanto conhecimento técnico, é conhecimento interpessoal. Quem quer ser mentor, querendo ou não, tem que ter paciência e o que infelizmente, vemos com frequência na nossa comunidade: pessoas **escrotas**. 

Acredito que não tem meio termo nisso. O mentor tem o papel de "pegar na mão" e explicar o caminho mais correto. A pessoa tem que ser capaz de lidar com N tipos de pessoas, das mais inteligentes às mais "lerdinhas". Com português impecável e CoM pOrTuGuêS MoDeRNo :laughing: 

Então se pudesse haver uma equipe para avaliar o perfil da pessoa, talvez uma conversa por hangout? Técnica é relativamente fácil avaliar, já perfil é mais complicado.

Quem sabe também cada mentor ter um feedback dos aprendizes? Dá para usar ferramentas como o Typeform para coletar mensalmente o desempenho do mentor. Dessa maneira, tem um controle de qualidade. E caso algum mentor esteja sendo escroto, acaba transparecendo nas respostas. E claro, a identidade nas respostas devem ser anônima.

@woliveiras mano, relaxa! eu vim do Cobol e AS/400 pra front :laughing:

**Comment by danilovaz** [comment]
@woliveiras então, o mentor pode disponibilizar um canal que seja mais privado, para que os aprendizes possam entrar em contato. Porém, seria interessante uma cultura de incentivar que perguntas por mais noobs que possam parecer, sejam perguntadas publicamente. Assim montamos uma comunidade madura, que sabe interpretar e conviver com perguntas dos mais variados níveis.

**Comment by LFeh** [comment]
:clap: :clap: :clap: :clap: 

Sensacional a movimentação de vocês para o desenvolvimento desse projeto.

É isso que esperamos com a criação desse repo no Github, que pessoas talentosas e proativas se unam para criar coisas boas para a comunidade.

Devem saber que estou atolado com 1 milhão de projetos e não posso me comprometer com mais nada. Mas estou disposto a ajudar conforme for possível.

**Comment by baltazarparra** [comment]
@woliveiras bora juntar mano, cria o repo lá e vamos dar vida a ideia.

**Comment by woliveiras** [comment]
Feito @baltazarparra! https://github.com/training-center/mentoria

Vou abrir umas issues pra gente conversar por lá os tópicos que levantamos

**Comment by LFeh** [comment]
Boa sorte caras ;)
